File: src/main/java/com/vimeo/networking/model/appconfiguration/AppConfiguration.java
package com.vimeo.networking.model.appconfiguration;

/**
 * Created by vennk on 5/20/15.
 */
public class AppConfiguration {
    
    public FacebookConfiguration facebook;
    public ApiConfiguration api;
    public FeaturesConfiguration features;


    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        AppConfiguration that = (AppConfiguration) o;

        if (facebook != null ? !facebook.equals(that.facebook) : that.facebook != null) {
            return false;
        }
        if (api != null ? !api.equals(that.api) : that.api != null) {
            return false;
        }
        return !(features != null ? !features.equals(that.features) : that.features != null);

    }

    @Override
    public int hashCode() {
        int result = facebook != null ? facebook.hashCode() : 0;
        result = 31 * result + (api != null ? api.hashCode() : 0);
        result = 31 * result + (features != null ? features.hashCode() : 0);
        return result;
    }
}

File: models/src/main/java/com/vimeo/networking2/AppConfiguration.kt
package com.vimeo.networking2

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.vimeo.networking2.annotations.Internal

/**
 * An object returned from the configs endpoint. This is a way for the API to specify configuration for our application.
 *
 * @param api API configuration data.
 * @param facebook Facebook API configuration data.
 * @param features Various feature configuration data.
 * @param live Various live streaming configuration data.
 */
@JsonClass(generateAdapter = true)
data class AppConfiguration(

    @Json(name = "api")
    val api: ApiConfiguration? = null,

    @Internal
    @Json(name = "facebook")
    val facebook: FacebookConfiguration? = null,

    @Internal
    @Json(name = "features")
    val features: FeaturesConfiguration? = null,

    @Internal
    @Json(name = "live")
    val live: LiveConfiguration? = null
)
